### DISCUSSION TITLE
Is it possible to change the ISP default DNS servers?

### DISCUSSION THREAD

**Discussion by vetalpeo** [Salamek/huawei-lte-api#196] [Q&A]
I would like to change my ISP DNS to cloudflare or google, but I can't find out how to do it. Anybody can help?

**Comment by Salamek** [comment]
My guess is that creating new profile using 

https://github.com/Salamek/huawei-lte-api/blob/1a28bbf96eec2480b04e03b46584f67ffeb9dc11/huawei_lte_api/api/DialUp.py#L55

and setting (needs to be implemented since create_profile is not using those values):

```python
'DnsIsStatic': '1',
'PrimaryDns': '8.8.8.8',
'SecondaryDns': '1.1.1.1',
```

in the call and then setting newly created profile as default using 
https://github.com/Salamek/huawei-lte-api/blob/1a28bbf96eec2480b04e03b46584f67ffeb9dc11/huawei_lte_api/api/DialUp.py#L41

should do the trick... 
but it is untested and i strongly recommend to backup any settings for your router/provider before doing this.

**Reply by vetalpeo** [reply]
Thanks, I will try it.
